src/test/java/io/cdap/plugin/servicenow/source/ServiceNowMultiSourceConfigTest.java

```diff
@@ -0,0 +1,180 @@
+  }
+
+  @Test
+  public void testValidateTableNames() {
+    MockFailureCollector mockFailureCollector = new MockFailureCollector("Stage Name");
+    serviceNowMultiSourceConfig.validateTableNames(mockFailureCollector);
+    Assert.assertEquals(0, mockFailureCollector.getValidationFailures().size());
+  }
+
+
+  @Test
+  public void testValidateTableNamesWhenTableNamesAreEmpty() {
+    serviceNowMultiSourceConfig =
+      ServiceNowSourceConfigHelper.newConfigBuilder()
+        .setReferenceName("Reference Name")
+        .setRestApiEndpoint(REST_API_ENDPOINT)
+        .setUser(USER)
+        .setPassword(PASSWORD)
+        .setClientId(CLIENT_ID)
+        .setClientSecret(CLIENT_SECRET)
+        .setTableNames("")
+        .setValueType("Actual")
+        .setStartDate("2021-12-30")
+        .setEndDate("2021-12-31")
+        .setTableNameField("tablename")
+        .buildMultiSource();
+    MockFailureCollector mockFailureCollector = new MockFailureCollector("Stage Name");
+    serviceNowMultiSourceConfig.validateTableNames(mockFailureCollector);
+    List<ValidationFailure> validationFailures = mockFailureCollector.getValidationFailures();
+    Assert.assertEquals(1, validationFailures.size());
+    ValidationFailure getResult = validationFailures.get(0);
+    List<ValidationFailure.Cause> causes = getResult.getCauses();
+    Assert.assertEquals(1, causes.size());
+    Assert.assertEquals("Table names must be specified.", getResult.getMessage());
+    Assert.assertEquals("Stage Name", getResult.getCorrectiveAction());
+    Assert.assertEquals("Table names must be specified. Stage Name", getResult.getFullMessage());
+    Assert.assertEquals("tableNames", causes.get(0).getAttributes().get("stageConfig"));
+  }
+
+  @Test
+  public void testValidateTableNamesWhenTableHasNoData() {
+    serviceNowMultiSourceConfig =
+      ServiceNowSourceConfigHelper.newConfigBuilder()
+        .setReferenceName("Reference Name")
+        .setRestApiEndpoint(REST_API_ENDPOINT)
+        .setUser(USER)
+        .setPassword(PASSWORD)
+        .setClientId(CLIENT_ID)
+        .setClientSecret(CLIENT_SECRET)
+        .setTableNames("clm_contract_history")
+        .setValueType("Actual")
+        .setStartDate("2021-12-30")
+        .setEndDate("2021-12-31")
+        .setTableNameField("tablename")
+        .buildMultiSource();
+    MockFailureCollector mockFailureCollector = new MockFailureCollector("Stage Name");
+    serviceNowMultiSourceConfig.validateTableNames(mockFailureCollector);
+    List<ValidationFailure> validationFailures = mockFailureCollector.getValidationFailures();
+    Assert.assertEquals(1, validationFailures.size());
+    ValidationFailure getResult = validationFailures.get(0);
+    List<ValidationFailure.Cause> causes = getResult.getCauses();
+    Assert.assertEquals(0, causes.size());
+    Assert.assertEquals("Table: clm_contract_history is empty.", getResult.getMessage());
```

Comment:
> Is this table always going to be empty on every ServiceNow instance?

Yes, by default, whenever a new instance is requested, this table is empty.


> General comment for all tests in this PR: instead of asking the user to provide ServiceNow credentials and connecting to a real instance, prefer to mock out API calls in the test.```

I've removed all instances wherever the tests required the user to specify ServiceNow credentials. [d5a1fe](https://github.com/data-integrations/servicenow-plugins/pull/8/commits/d5a1fe372863518cbd195206cb8ddcf0aac2e4a4)